### ISSUE TITLE
Error: SSL=no doesn't exist

### ISSUE THREAD

**Issue by cheluu** [interbrite/letsencrypt-vesta#43]
Hi, i try yo encrypt my site and i get Error: SSL=no doesn't exist

I'm on ubuntu in digitalocean 

```
 IMPORTANT NOTES:
 - Congratulations! Your certificate and chain have been saved at
   /etc/letsencrypt/live/dxequipamientos.ml/fullchain.pem. Your cert
   will expire on 2017-03-10. To obtain a new or tweaked version of
   this certificate in the future, simply run certbot-auto again. To
   non-interactively renew *all* of your certificates, run
   "certbot-auto renew"
 - If you like Certbot, please consider supporting our work by:

   Donating to ISRG / Let's Encrypt:   https://letsencrypt.org/donate
   Donating to EFF:                    https://eff.org/donate-le

Error: SSL=no doesn't exist
 * Reloading web server apache2     
```                                             *


Thank u for any help ! 👍 





SOLVED: https://github.com/interbrite/letsencrypt-vesta/pull/40/commits/d1cf5088841da91eb65100c4823e1d4624962bd1

FORGIVE ME

**Comment by elsker** [comment]
Hello,
I'm having the same error when trying to install a SSL certificate. I run with Vesta/nginx/Centos7. I've followed the step from readme file.

'
Your cert
   will expire on 2017-04-02. To obtain a new or tweaked version of
   this certificate in the future, simply run certbot-auto again. To
   non-interactively renew *all* of your certificates, run
   "certbot-auto renew"
 - If you like Certbot, please consider supporting our work by:

   Donating to ISRG / Let's Encrypt:   https://letsencrypt.org/donate
   Donating to EFF:                    https://eff.org/donate-le

Error: SSL=no doesn't exist
Redirecting to /bin/systemctl reload  nginx.service
'

**Comment by harkalygergo** [comment]
Same problem, I tried with admin user and subuser too on Ubuntu 16.04 at DigitalOcean with VestaCP.

```
sudo letsencrypt-vesta admin XXX
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Starting new HTTPS connection (1): acme-v01.api.letsencrypt.org
Renewing an existing certificate
Performing the following challenges:
http-01 challenge for polygons.info
http-01 challenge for www.XXX
http-01 challenge for XXX.XXX
Using the webroot path /etc/letsencrypt/webroot for all unmatched domains.
Waiting for verification...
Cleaning up challenges
Generating key (2048 bits): /etc/letsencrypt/keys/0006_key-certbot.pem
Creating CSR: /etc/letsencrypt/csr/0006_csr-certbot.pem

IMPORTANT NOTES:
 - Congratulations! Your certificate and chain have been saved at
   /etc/letsencrypt/live/XXX/fullchain.pem. Your cert will
   expire on 2017-04-03. To obtain a new or tweaked version of this
   certificate in the future, simply run certbot-auto again. To
   non-interactively renew *all* of your certificates, run
   "certbot-auto renew"
 - If you like Certbot, please consider supporting our work by:

   Donating to ISRG / Let's Encrypt:   https://letsencrypt.org/donate
   Donating to EFF:                    https://eff.org/donate-le

Error: SSL=no doesn't exist

```

**Comment by nedix** [comment]
I'm running the latest version of certbot, letsencrypt-vesta and VestaCP 0.9.8-17 and have done a full reboot since updating.

I'm experiencing the same issue with SSL=no doesn't exist and noticed there now is a Letsencrypt button in Vesta which gives me: Error: Invalid response from http://[domain]/.well-known/acme-challenge/xxxx_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx: \

I'm suspicious this is a permissions issue but can't figure out exactly where it's coming from.

It looks like all the certificate files are located in the /home/[user]/web/[domain]/private/tmp.xxxxxxxxx folder with the folder permissions being 700 and file permissions 644.

**Comment by elsker** [comment]
I've managed to make it work.
Just modify the file /usr/local/letsencrypt-vesta/letsencrypt-vesta like here:
[https://github.com/interbrite/letsencrypt-vesta/pull/40/commits/d1cf5088841da91eb65100c4823e1d4624962bd1](https://github.com/interbrite/letsencrypt-vesta/pull/40/commits/d1cf5088841da91eb65100c4823e1d4624962bd1)

**Comment by nedix** [comment]
Darn, thought that patch was already merged with master. That did the trick.

**Comment by elsker** [comment]
Yeah me too. But now I have to wait one week because I've already reached the max limit (5) for same domain. :(

**Comment by nedix** [comment]
Had the same issue for one of my domains. Found that all the previous attempts left tmp folders in /home/[user]/web/[domain]/private/ containing cert, CA and key files which I could manually enter via VestaCP.

**Comment by cyfocus** [comment]
![fix-centos7-letsencrypt-vestacp](https://cloud.githubusercontent.com/assets/11606451/21707576/5a42dfc4-d39e-11e6-93e9-660db8de8294.JPG)
I fix mine through VestaCP control panel and click the check box to enable SSL Support and Letsencrypt Support and click Save.  Then SSH to your server and type #letsencrypt-vesta USER DOMAIN.  It should work.

**Comment by harkalygergo** [comment]
I tried elsker's answer (https://github.com/interbrite/letsencrypt-vesta/issues/43#issuecomment-270739585) yesterday on two servers, but both of them stopped. On first  Nginx doesn't want to work, on the other Apache.

**Comment by joomlagate** [comment]
Great! Today I met the same "Error: SSL=no doesn't exist" issue on Ubuntu 16.04, and then fixed this with the patch in https://github.com/interbrite/letsencrypt-vesta/pull/40/files .

Then I ran the command "letsencrypt-vesta USER DOMAIN" again, there is no errors and my SSL certificate was generated and implemtned automatically! Now my website can be accessed via HTTPS!

**Comment by fabianborg** [comment]
I have carried the correction to the `/usr/local/letsencrypt-vesta/letsencrypt-vesta` as per the patch [here](https://github.com/interbrite/letsencrypt-vesta/pull/40/commits/d1cf5088841da91eb65100c4823e1d4624962bd1) and the domain generation worked, however when I run the `letsencrypt-vesta -a 60 USERNAME DOMAIN` (USERNAME DOMAIN are replaced with the real ones), I am getting the following error 

<pre>
/usr/local/bin/letsencrypt-vesta: line 272: syntax error near unexpected token `done'
/usr/local/bin/letsencrypt-vesta: line 272: `    done'
</pre>

However I am not getting the renewal notice confirmation hereunder as it used to.

<pre>
warning: commands will be executed using /bin/sh
job 1 at Sat Jul  1 16:57:00 2017
Renewal scheduled for 60 days.
</pre> 

How can I solve this?

**Comment by listabite** [comment]
Thank you so much! I had the same problem, now its working.. ;)
